tests/backstop/scenarios.js

```diff
@@ -1,8 +1,84 @@
 const storybookHost = require( './detect-storybook-host' );
 const rootURL = `${ storybookHost }iframe.html?id=`;
 const storybookStories = require( '../../.storybook/storybook-data' );
+const filePaths = require( '../../.storybook/main' );
+const glob = require( 'glob' );
+const fs = require( 'fs' );
+const parser = require( '@babel/parser' );
+const traverse = require( '@babel/traverse' ).default;
+const csf = require( '@componentdriven/csf' );
+const kebabCase = require( 'lodash/kebabCase' );
 
-module.exports = storybookStories.map( ( story ) => {
+const newBackstopTests = [];
+
+const storyFiles =
+	// remove relative parent ..
+	glob.sync( filePaths.stories[ 1 ].slice( 1 ) );
+
+storyFiles.forEach( ( storyFile ) => {
+	const code = fs.readFileSync( storyFile ).toString();
+	const ast = parser.parse( code, {
+		sourceType: 'module',
+		plugins: [ 'jsx' ],
+	} );
+
+	const stories = {};
+	let defaultTitle = '';
+
+	traverse( ast, {
+		ExportDefaultDeclaration: ( { node } ) => {
+			const properties = {};
+			node.declaration.properties.forEach( ( property ) => {
+				properties[ property.key.name ] =
+					property.value.value || property.value.name;
+			} );
+
+			defaultTitle = ( properties && properties.title ) || '';
+		},
+		AssignmentExpression: ( { node } ) => {
+			let nodeValue = '';
+			if (
+				node.right.type === 'StringLiteral' &&
+				node.left.property.name === 'storyName'
+			) {
+				nodeValue = node.right.value;
+			} else if ( node.right.type === 'ObjectExpression' ) {
+				nodeValue = {};
+				node.right.properties.forEach( ( property ) => {
+					nodeValue[ property.key.name ] = property.value.value;
+				} );
+			}
+
+			if ( ! stories[ node.left.object.name ] ) {
+				stories[ node.left.object.name ] = {};
+			}
+
+			stories[ node.left.object.name ][
+				node.left.property.name
+			] = nodeValue;
+		},
+	} );
+
+	for ( const [ key, value ] of Object.entries( stories ) ) {
+		const storyID = csf.toId( defaultTitle, kebabCase( key ) ); // eslint-disable-line sitekit/acronym-case
+		if (
+			value &&
+			value.scenario &&
+			Object.keys( value.scenario ).length > 0 &&
+			// TODO - also support bool here
```

Comment:
I've tried doing this with no luck

I'm not sure whether there is a problem with `@babel/parser` or `@babel/traverse` but if I set `scenario = true` on a CSF story, then here the object has an empty string assigned to the scenario key :thinking: 

I can look into this more but I don't know anything about these libraries (apart from a high-level overview of what they do)

Reply:
The quickest thing to do would be just to make it be an empty object.  Then we can just check truthy here and spread in an empty object if no options are set :shrug:

Reply:
Yeah that was how it was initially defined but having `Story.scenario = {}` is a bit odd to enable a Backstop test as you mentioned so it would be nice to allow for `true` here as well. Not strictly necessary if it's too complicated but would be a nice to have for sure.

Reply:
I just don't much about these AST parsing packages.  I'd :heart:  to learn more but it might be out-of-scope of the ticket. I could spend an hour and see where I get? @aaemnnosttv